src/EVM/SymExec.hs

```diff
@@ -1117,6 +1119,23 @@ prettyBuf (ConcreteBuf "") = "Empty"
 prettyBuf (ConcreteBuf bs) = formatBinary bs
 prettyBuf b = internalError $ "Unexpected symbolic buffer:\n" <> T.unpack (formatExpr b)
 
+calldataFromCex :: App m => SMTCex -> Expr Buf -> Text -> [AbiType] -> m (Err ByteString)
+calldataFromCex cex buf funName types = do
+  let fullSig = (funName <> "(" <> T.intercalate "," (map abiTypeSolidity types) <> ")")
+      sigKeccak = BS.take 4 $ keccakSig $ encodeUtf8 fullSig
```

Comment:
Isn't one of the `take 4` redundant? Aren't you already taking the first 4 bytes already in `keccakSig`?

Reply:
Haha oh yes. Thanks, fixed!